Orange/widgets/regression/tests/test_owregressiontree.py

```diff
@@ -1,11 +1,48 @@
 # Test methods with long descriptive names can omit docstrings
 # pylint: disable=missing-docstring
 from Orange.widgets.regression.owregressiontree import OWRegressionTree
-from Orange.widgets.tests.base import WidgetTest, WidgetLearnerTestMixin
+from Orange.widgets.tests.base import (WidgetTest, WidgetLearnerTestMixin,
+                                       GuiToParam)
 
 
 class TestOWRegressionTree(WidgetTest, WidgetLearnerTestMixin):
     def setUp(self):
         self.widget = self.create_widget(OWRegressionTree,
                                          stored_settings={"auto_apply": False})
         self.init()
+
+        def combo_set_value(i, x):
+            x.activated.emit(i)
+            x.setCurrentIndex(i)
+
+        scores = [score[1] for score in self.widget.scores]
+        md_spin = self.widget.max_depth_spin[1]
+        mi_spin = self.widget.min_internal_spin[1]
+        ml_spin = self.widget.min_leaf_spin[1]
+        md_min_max = [md_spin.minimum(), md_spin.maximum()]
+        mi_min_max = [mi_spin.minimum(), mi_spin.maximum()]
+        ml_min_max = [ml_spin.minimum(), ml_spin.maximum()]
+        self.gui_to_params = [
+            GuiToParam('criterion', self.widget.score_combo,
+                       lambda x: scores[x.currentIndex()],
+                       combo_set_value, scores, list(range(len(scores)))),
+            GuiToParam('max_depth', md_spin, lambda x: x.value(),
+                       lambda i, x: x.setValue(i), md_min_max, md_min_max),
+            GuiToParam('min_samples_split', mi_spin, lambda x: x.value(),
+                       lambda i, x: x.setValue(i), mi_min_max, mi_min_max),
+            GuiToParam('min_samples_leaf', ml_spin, lambda x: x.value(),
+                       lambda i, x: x.setValue(i), ml_min_max, ml_min_max)]
+
+    def test_parameters_unchecked(self):
+        """Check learner and model for various values of all parameters
+        when pruning parameters are not checked
+        """
+        self.widget.max_depth_spin[0].setCheckState(False)
+        self.widget.min_internal_spin[0].setCheckState(False)
+        self.widget.min_leaf_spin[0].setCheckState(False)
+        for i, val, f in zip(range(1, 4), [None, 2, 1],
```

Comment:
I would rewrite this as a tuple of tuples,
 ((1, None, lambda x: None),
  (2, 2, lambda x:2),
  (3, 1, lambda x:1))
to make it more obvious which values go together.

After doing that, it looks like the third parameter (the function) just returns the value of second parameter. If this is the case,  could we just create the function inside for loop?